types/hapi/index.d.ts

```diff
@@ -3806,8 +3806,8 @@ export class Server extends Podium {
 /**
  *  User-extensible type for application specific state.
  */
- /* tslint:disable-next-line:no-empty-interface */
 export interface ApplicationState {
+    [key: string]: any,
```

Comment:
This type is intentionally left as an empty interface to allow you to use declaration merging to define the ApplicationState as it is in your application.

This actually reduces type safety rather than increases it.  

If you want your own custom state, in your application, you should redefine this, as it is shown in the test cases.